src/server/middlewares/connector-icons.ts

```diff
@@ -0,0 +1,46 @@
+import type {NextFunction, Request, Response} from '@gravity-ui/expresskit';
+import type {ApiActionConfig} from '@gravity-ui/gateway';
+import type {AppContext} from '@gravity-ui/nodekit';
+import merge from 'lodash/merge';
+
+import type {ConnectorIconData, ListConnectorIconsResponse} from '../../shared/schema/types';
+import {registry} from '../registry';
+import type {DatalensGatewaySchemas} from '../types/gateway';
+
+type ActionArgs = ApiActionConfig<AppContext, undefined, ListConnectorIconsResponse>;
+type GetAdditionalArgs = (req: Request, res: Response) => Partial<ActionArgs>;
+
+export function getConnectorIconsMiddleware({
+    getAdditionalArgs,
+}: {getAdditionalArgs?: GetAdditionalArgs} = {}) {
+    return async function (req: Request, res: Response, next: NextFunction) {
+        let icons: ConnectorIconData[] = [];
+
+        try {
+            req.ctx.log('REQUEST_CONNECTOR_ITEMS');
+            const {gatewayApi} = registry.getGatewayApi<DatalensGatewaySchemas>();
+            const actionArgs: ActionArgs = {
+                args: undefined,
```

Comment:
Yep, because of [current types](https://github.com/gravity-ui/gateway/blob/8dcba28561e1d636b96c74c00247ee3c29a61ccc/lib/models/common.ts#L38). `TRequestData` could be `undefined`, but you should specify it in any case because there is no `?` in type definition